matrix-meetings-bot/src/service/RoomMessageService.ts

```diff
@@ -367,6 +374,71 @@ export class RoomMessageService {
       notification += this.createRepetitionLine(lng, oldRruleText, true);
     }
 
+    for (const change of meetingChanges.occurrenceChanged) {
+      if (change.changeType === 'add' || change.changeType === 'update') {
+        const start: Date = parseICalDate(change.value.dtstart).toJSDate();
+        const end: Date = parseICalDate(change.value.dtend).toJSDate();
+
+        const range = dateRangeFormat(start, end, lng, timeZone, {
+          month: 'long',
+        });
+        notification += this.formatCurrent(
+          i18next.t(
+            'meeting.room.notification.changed.occurrence',
+            'A single meeting from a meeting series is moved to {{range}}',
+            { lng, range },
+          ),
+        );
+
+        const [prevStart, prevEnd] =
+          change.changeType === 'add'
+            ? [
+                parseICalDate(change.value.recurrenceId).toJSDate(),
+                parseICalDate(change.dtend).toJSDate(),
+              ]
+            : [
+                parseICalDate(change.oldValue.dtstart).toJSDate(),
+                parseICalDate(change.oldValue.dtend).toJSDate(),
+              ];
+
+        const prevRange = dateRangeFormat(prevStart, prevEnd, lng, timeZone, {
+          month: 'long',
+        });
+        notification += this.formatPrevious(
+          i18next.t(
+            'meeting.room.notification.changed.previous',
+            '(previously: {{value}})',
+            { lng, value: prevRange },
+          ),
+        );
+      } else if (
+        change.changeType === 'delete' ||
+        change.changeType === 'exdate'
+      ) {
+        const [start, end] =
+          change.changeType === 'delete'
+            ? [
+                parseICalDate(change.value.dtstart).toJSDate(),
+                parseICalDate(change.value.dtend).toJSDate(),
+              ]
+            : [
+                parseICalDate(change.dtstart).toJSDate(),
+                parseICalDate(change.dtend).toJSDate(),
+              ];
+
+        const range = dateRangeFormat(start, end, lng, timeZone, {
+          month: 'long',
+        });
+        notification += this.formatCurrent(
+          i18next.t(
+            'meeting.room.notification.changed.occurrence_deleted',
+            'A single meeting from a meeting series on {{range}} is deleted',
+            { lng, range },
+          ),
+        );
```

Comment:
I wonder why the `dateRangeFormat` is called manually and not via `i18next` as `{{range, daterange}}` [like we do it in the meetings widget](https://github.com/nordeck/matrix-meetings/blob/0dcc0950ad3616a8f53d69e873aee9a12dcdbbf3/matrix-meetings-widget/src/components/common/DateTimePickers/CalendarWeekPicker.tsx#L88-L94). I guess I oversaw that when reviewing https://github.com/nordeck/matrix-meetings/pull/298.

I can assist if you need help to set it up probably. It would also be a good opportunity to get rid of `moment-timezone` and just use `i18next` to format the dates [here](https://github.com/nordeck/matrix-meetings/blob/0dcc0950ad3616a8f53d69e873aee9a12dcdbbf3/matrix-meetings-bot/src/service/RoomMessageService.ts#L323C43-L342) (or in the new code in case we opt to rewrite it).

Reply:
date range formatting is alled via `i18next` now, should be fine